mapservutil.c

```diff
@@ -201,11 +203,15 @@ mapObj *msCGILoadMap(mapservObj *mapserv, configObj *config)
 
   /* validate ms_mapfile if tainted */
   if(ms_mapfile_tainted == MS_TRUE) {
+    if(ms_map_pattern == NULL) { // can't go any further, bail
+      msSetError(MS_WEBERR, "Required configuration value MS_MAP_PATTERN not set.", "msCGILoadMap()");
```

Comment:
As I understand with, we can't any more pass it with an environment variable.
Is it correct?
Is it wanted?

Reply:
@sbrunner, I'm not sure I follow your question. What 'it' are you referring to? MS_MAP_PATTERN is set in the config file - basically the equivalent of an environment variable.

Reply:
Yes, But in my tests I set it as an environment variable but not in the config bile and I get this error...

Reply:
@sdlime - I'm guessing the question is can MS_MAP_PATTERN be set in either the CONFIG file _and_ also via a system environment variable?

Reply:
It should be loadable from either. We're using the GDAL configuration options support to manage portions of the configuration data. Specifically, all elements set in the config file ENV section are pushed into that space using CPLSetConfigOption(). Then they are retrieved as necessary using CPLGetConfigOption(). My understanding from @rouault, is that CPLGetConfigOption() also checks the environment if an option is not available in the config. Sounds like your test contradicts that?

Reply:
I confirm that the environment is set, and I get the error:
```
mod_fcgid: stderr: msCGILoadMap(): Web application error. Required configuration value MS_MAP_PATTERN not set.
```

Reply:
Hopefully Even can confirm behavior. Assuming your tests are correct I guess the question is do we want to support both methods. It certainly wasn't a design goal of mine, rather a feature that I thought we got for free via GDAL. Intuitively it seems like NOT supporting multiple methods would be preferred since the idea behind the config file was to streamline configuration across platforms.  Is there a use case for supporting both methods? If so, which variables?

Reply:
I could see some use for variables which are common for Mapserver and something else but are there any? GDAL, Proj, curl?

Reply:
I was using the environment variable but if it's not working anymore it's not an issue, I will change it in our side :-)
Thanks for the clarifications :-)

Reply:
@sbrunner, cool. Now you'll have one environment variable for the config file and then store the detail there. It's a change but not too bad. --Steve